sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -72,18 +94,14 @@ trait DateTimeFormatterHelper {
   // DateTimeParseException will address by the caller side.
   protected def checkDiffResult[T](
       s: String, legacyParseFunc: String => T): PartialFunction[Throwable, T] = {
-    case e: DateTimeParseException if SQLConf.get.legacyTimeParserPolicy == EXCEPTION =>
-      val res = try {
-        Some(legacyParseFunc(s))
-      } catch {
-        case _: Throwable => None
-      }
-      if (res.nonEmpty) {
+    case e: DateTimeException if SQLConf.get.legacyTimeParserPolicy == EXCEPTION =>
```

Comment:
when the field value exceeds the valid range, the JDK throws `DateTimeException` too, we should catch it. `DateTimeParseException` extends `DateTimeException`